File: CSharp/NetCubeSolver/ExactCover/CoverNode.cs
namespace NetCubeSolver.ExactCover;

public class CoverNode
{

    public int ColumnId;
    public int Size;
    public bool IsColumn;

    public CoverNode Left;
    public CoverNode Right;
    public CoverNode Up;
    public CoverNode Down;
    public CoverNode Column;
    
    public CoverNode()
    {
        IsColumn = false;
        ColumnId = -1;
        Size = 0;
    }

    public CoverNode(int columnId)
    {
        ColumnId = columnId;
        IsColumn = true;
        Size = 0;
    }
}

File: Kotlin/app/src/main/kotlin/ca/jwolf/cubesolver/datamodels/CoverNode.kt
package ca.jwolf.cubesolver.datamodels

class CoverNode( val columnId: Int) {

    val isColumn = columnId >= 0
    var size = 0

    constructor() : this(-1) {
    }

    var left : CoverNode? = null
    var right: CoverNode? = null
    var up: CoverNode? = null
    var down: CoverNode? = null
    var column: CoverNode? = null

}